### ISSUE TITLE
Presolver crash instead of return unfeasible solution on java

### ISSUE THREAD

**Issue by jfvg71** [google/or-tools#1104]
Hi everybody, while writing an example I found myself with the following crash of the library:

`WARNING: Logging before InitGoogleLogging() is written to STDERR
F0302 15:03:40.557718  7833 cp_model_presolve.cc:107] Check failed: !DomainIsEmpty(ref) 
*** Check failure stack trace: ***`

The following example, when executed in the JVM, makes it crash.

```java import java.util.ArrayList;
import java.util.Arrays;

import com.google.ortools.sat.CpModel;
import com.google.ortools.sat.CpSolver;
import com.google.ortools.sat.CpSolverStatus;
import com.google.ortools.sat.IntVar;

public class StudentRostering {
        static {
	    System.loadLibrary("jniortools");
	 }
	private static void solve() {
		// This program tries to find an optimal assignment of students to shifts
		// (2 shifts per day, for 10 days), subject to some constraints (see below).
		// Each student can request to be assigned to specific shifts and can
		// request shifts that can't work
		// The optimal assignment maximizes the number of fulfilled shift requests.
		int num_students = 10;
		int num_shifts = 2;
		int num_days = 10;
		int[][][] shift_requests = {
				{ { 1, 1 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 },
						{ 0, 0 } } };

		int[][][] no_shifts = {
				{ { 1, 1 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 1, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 } },
				{ { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 },
						{ 0, 0 } } };

		// Create the model.
		CpModel model = new CpModel();

		// Creates shift variables.
		// shifts[(n, d, s)]: student 'n' works shift 's' on day 'd'.
		IntVar[][][] shifts = new IntVar[num_students][num_days][num_shifts];
		for (int i = 0; i < num_students; i++)
			for (int j = 0; j < num_days; j++)
				for (int k = 0; k < num_shifts; k++)
					shifts[i][j][k] = model.newBoolVar(String.format("Shift_%d_%d_%d", i, j, k));

		// Each shift is assigned to exactly two students.
		for (int j = 0; j < num_days; j++)
			for (int k = 0; k < num_shifts; k++) {
				ArrayList<IntVar> daily_shifts = new ArrayList<>();
				for (int i = 0; i < num_students; i++)
					daily_shifts.add(shifts[i][j][k]);
				model.addLinearSumEqual(daily_shifts.toArray(new IntVar[1]), 2);
			}

		// Each student works at most one shift per day.
		for (int i = 0; i < num_students; i++) {
			for (int j = 0; j < num_days; j++) {
				ArrayList<IntVar> worked_shifts = new ArrayList<>();
				for (int k = 0; k < num_shifts; k++)
					worked_shifts.add(shifts[i][j][k]);
				model.addLinearSum(worked_shifts.toArray(new IntVar[1]), 0, 1);
			}
		}

		// min_shifts_assigned is the largest integer such that every student can be
		// assigned at least that number of shifts.
		int min_shifts_per_student = (2 * num_shifts * num_days) / num_students;
		int max_shifts_per_student = min_shifts_per_student + 1;
		for (int i = 0; i < num_students; i++) {
			ArrayList<IntVar> num_shifts_worked = new ArrayList<>();
			for (int j = 0; j < num_days; j++)
				for (int k = 0; k < num_shifts; k++)
					num_shifts_worked.add(shifts[i][j][k]);
			model.addLinearSum(num_shifts_worked.toArray(new IntVar[1]), min_shifts_per_student,
					max_shifts_per_student);
		}

		// preventing students to work in night shifts less that
		// min_shifts_per_student/2 and more that min_shifts_per_student/2 + 1
		for (int i = 0; i < num_students; i++) {
			ArrayList<IntVar> num_nightshifts_worked = new ArrayList<>();
			for (int j = 0; j < num_days; j++)
				num_nightshifts_worked.add(shifts[i][j][1]);// night shifts only
			model.addLinearSum(num_nightshifts_worked.toArray(new IntVar[1]), min_shifts_per_student / 2,
					min_shifts_per_student / 2 + 1);
		}

		// students need to have at less a day off between shifts
		for (int i = 0; i < num_students; i++) {
			for (int j = 0; j < num_days - 1; j++) {
				ArrayList<IntVar> shifts_consworked = new ArrayList<>();
				for (int k = 0; k < num_shifts; k++) {
					shifts_consworked.add(shifts[i][j][k]);
					shifts_consworked.add(shifts[i][j + 1][k]);
				}
				model.addLinearSum(shifts_consworked.toArray(new IntVar[1]), 0, 1);
			}
		}

		// Preventing assign a shift to student that can't work in this shift
		for (int i = 0; i < num_students; i++) {
			for (int j = 0; j < num_days; j++)
				for (int k = 0; k < num_shifts; k++)
					if (no_shifts[i][j][k] == 1)
						model.addLessOrEqual(shifts[i][j][k], 0);
		}

		// The optimal assignment maximizes the number of fulfilled shift requests.
		for (int i = 0; i < num_students; i++) {
			ArrayList<IntVar> nurse_shifts = new ArrayList<>();
			ArrayList<Integer> nurse_request = new ArrayList<>();
			for (int j = 0; j < num_days; j++)
				for (int k = 0; k < num_shifts; k++) {
					nurse_shifts.add(shifts[i][j][k]);
					nurse_request.add(shift_requests[i][j][k]);
				}
			model.maximizeScalProd(nurse_shifts.toArray(new IntVar[1]),
					nurse_request.stream().mapToInt(n -> n).toArray());
		}

		// Creates the solver and solve.
		CpSolver solver = new CpSolver();
		CpSolverStatus status = solver.solve(model);
		if (status == CpSolverStatus.INFEASIBLE) {
			System.out.println("Unfeasible model, please review the data!");
			System.exit(1);
		}
		int students[] = new int[num_students];

		for (int j = 0; j < num_days; j++) {
			System.out.println("Day " + j);
			for (int i = 0; i < num_students; i++)
				for (int k = 0; k < num_shifts; k++)
					if (solver.value(shifts[i][j][k]) == 1) {
						students[i]++;
						if (shift_requests[i][j][k] == 1)
							System.out.println("Student " + i + " works shift " + k + " (requested).");
						else
							System.out.println("Student " + i + " works shift " + k + " (not requested).");
					}
			System.out.println();
		}
		int night_shifts[] = new int[num_students];
		for (int i = 0; i < num_students; i++) {
			System.out.printf("%02d: ", i);
			for (int j = 0; j < num_days; j++) {
				boolean off = true;
				for (int k = 0; k < num_shifts; k++)
					if (solver.value(shifts[i][j][k]) == 1) {
						off = false;
						if (k == 0)
							System.out.print("d ");
						else {
							night_shifts[i]++;
							System.out.print("n ");
						}
					}
				if (off)
					System.out.print("o ");
			}
			System.out.println();
		}

		System.out.println("Working shifts per student: " + Arrays.toString(students));
		System.out.println("Working night shifts per student: " + Arrays.toString(night_shifts));
		// Statistics.
		int count = 0;
		for (int[][] days : shift_requests)
			for (int[] shift : days)
				for (int req : shift)
					if (req == 1)
						count += 1;

		System.out.println();
		System.out.println("Statistics");
		System.out.println("  - Number of shift requests met = " + solver.objectiveValue() + " (out of " + count + ")");
		System.out.println("  - Branches: " + solver.numBranches());
		System.out.println("  - Conflicts: " + solver.numConflicts());
		System.out.println("  - wall time       : " + solver.wallTime());

	}

	public static void main(String[] args) throws Exception {
		StudentRostering.solve();
	}

}
```
If one removes the constraints that force to leave at least a day off between assignments, i. e.:

```java // students need to have at less a day off between shifts
for (int i = 0; i < num_students; i++) {
	for (int j = 0; j < num_days - 1; j++) {
		ArrayList<IntVar> shifts_consworked = new ArrayList<>();
		for (int k = 0; k < num_shifts; k++) {
			shifts_consworked.add(shifts[i][j][k]);
			shifts_consworked.add(shifts[i][j + 1][k]);
		}
		model.addLinearSum(shifts_consworked.toArray(new IntVar[1]), 0, 1);
	}
}
```
The the program runs properly and prints the output:

`Unfeasible model, please review the data!`

**Comment by lperron** [comment]
This is already fixed on master.
I just ran the example.

**Comment by jfvg71** [comment]
I'm having a problem while compiling (make third_party) the latest version in the master branch (apparently a problem with abseil library):
`CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:33 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:44 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:58 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:75 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:93 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:105 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:140 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:167 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:304 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/base/CMakeLists.txt:398 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/algorithm/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/algorithm/CMakeLists.txt:37 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:20 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:26 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:48 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:110 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:125 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:195 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:228 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:263 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:297 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:329 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:353 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:418 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:440 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:467 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:491 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:502 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:512 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:521 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:542 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:555 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:566 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/container/CMakeLists.txt:624 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:33 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:71 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:87 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:122 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:149 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:180 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:194 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/debugging/CMakeLists.txt:298 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/hash/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/hash/CMakeLists.txt:85 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/memory/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/meta/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/meta/CMakeLists.txt:44 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/numeric/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/numeric/CMakeLists.txt:52 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/strings/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/strings/CMakeLists.txt:69 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/strings/CMakeLists.txt:354 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/strings/CMakeLists.txt:366 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/synchronization/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/synchronization/CMakeLists.txt:33 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/time/CMakeLists.txt:17 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/time/CMakeLists.txt:44 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/time/CMakeLists.txt:56 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:16 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:32 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:45 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:112 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:170 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:189 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:207 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/types/CMakeLists.txt:260 (absl_cc_library)


CMake Error at CMake/AbseilHelpers.cmake:105 (list):
  list does not recognize sub-command FILTER
Call Stack (most recent call first):
  absl/utility/CMakeLists.txt:17 (absl_cc_library)


-- Configuring incomplete, errors occurred!
See also "/home/jfvg/Downloads/or-tools/dependencies/sources/abseil-cpp-93d155b/build_cmake/CMakeFiles/CMakeOutput.log".
See also "/home/jfvg/Downloads/or-tools/dependencies/sources/abseil-cpp-93d155b/build_cmake/CMakeFiles/CMakeError.log".
makefiles/Makefile.third_party.unix.mk:345: recipe for target 'dependencies/install/lib/libabsl.so' failed
make: *** [dependencies/install/lib/libabsl.so] Error 1
`
Do you have any idea?

**Comment by lperron** [comment]
Can you check the cmake version?
Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00

**Comment by Mizux** [comment]
FILTER has been introduced in CMake 3.6
> The [list()](https://cmake.org/cmake/help/v3.6/command/list.html) command gained a `FILTER` sub-command to filter list elements by regular expression.
see: https://cmake.org/cmake/help/v3.6/release/3.6.html#commands

BTW you should use the Makefile-based build since CMake-based build does not support java yet...

**Comment by jfvg71** [comment]
```
cmake --version
cmake version 3.5.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

**Comment by Mizux** [comment]
note: Abseil-cpp has fixed it internally CL 235725207 should be soon available out there -> but will need to bump the SHA1 used by or-tools..

**Comment by jfvg71** [comment]
Problem solved with the newest version of CMake.
